src/ip_notifier.cpp

```diff
@@ -74,10 +222,29 @@ namespace libtorrent
 	void ip_change_notifier::async_wait(std::function<void(error_code const&)> cb)
 	{
 #if defined TORRENT_BUILD_SIMULATOR
-		TORRENT_UNUSED(cb);
+		// TODO: simulator support
+		cb(make_error_code(boost::system::errc::not_supported));
 #elif TORRENT_USE_NETLINK
+		using namespace std::placeholders;
 		m_socket.async_receive(boost::asio::buffer(m_buf)
 			, std::bind(&ip_change_notifier::on_notify, this, _1, _2, cb));
+#elif TORRENT_USE_SYSTEMCONFIGURATION
+		m_cb = std::move(cb);
```

Comment:
It would be better to capture the callback directly rather than assigning it to a member and capturing `this`. That way if there are multiple callers they won't clobber each other's callbacks.